subprojects/core/src/main/java/org/gradle/initialization/ParallelismBuildOptionFactory.java

```diff
@@ -75,15 +76,11 @@ public void applyTo(String value, ParallelismConfiguration settings, Origin orig
         }
 
         private void handleInvalidValue(Origin origin, String value) {
-            origin.handleInvalidValue(createGradlePropertyFailureMessage(value), createCommandLineFailureMessage(value));
-        }
-
-        private String createGradlePropertyFailureMessage(String value) {
-            return String.format(String.format("Value '%s' given for %s Gradle property is invalid (must be a positive, non-zero, integer)", value, gradleProperty));
-        }
-
-        private String createCommandLineFailureMessage(String value) {
-            return String.format("Argument value '%s' given for --%s option is invalid (must be a positive, non-zero, integer)", value, LONG_OPTION);
+            if (origin == Origin.COMMAND_LINE) {
```

Comment:
:x: The handling of the `if`/`else if` should rather live in `Origin` as you'll have to have the same logic in multiple places.